frontend/src/components/common/SyncButton.vue

```diff
@@ -1,38 +1,48 @@
 <script setup lang="ts">
-import { ref, onMounted } from 'vue';
+import { ref } from 'vue';
 
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
-import { Button, Dialog} from '@/lib/primevue';
-import { useObjectStore } from '@/store';
+import { Button, Dialog, useToast } from '@/lib/primevue';
+import { useObjectStore, useBucketStore } from '@/store';
 
-import type { Ref } from 'vue';
-import type { COMSObject } from '@/types';
 
 // Props
 type Props = {
+  type: string;
   id: string;
+  name: string;
 };
```

Comment:
Is there a reason why we didn't do props like this instead?
```ts
type Props = {
  bucketId?: string,
  objectId?: string
};
```
Forcing a name dependency upstream as a required prop doesn't appear to be a great idea; the component should be the one being able to figure out that information itself. Also if we want to be doing an explicit type string instead, this should be defined as an enum type, not as two constant strings.